iree/compiler/Dialect/HAL/IR/LoweringConfig.h

```diff
@@ -27,6 +27,30 @@
 namespace mlir {
 namespace iree_compiler {
 
+namespace IREE {
+namespace HAL {
+/// Struct that for a given hal.target.executable defines how it is translated.
+// TODO(ravishankarm): This could also be converted to an attribute on the
```

Comment:
yeah we should chat about converting this to an attribute - I'll be (hopefully, after 10 other things) getting to the point where we can plumb the attribute through hopefully in a few weeks - if it has to start as just a loose DictionaryAttr or ArrayAttr that's mapped to this that's probably fine but we'd want to get some basic structure on the top level and avoid the duplication into a c++ class asap - otherwise we end up in the classic plumbing case of adding something new requires changing all of this and all backends using it - and versioning gets tricky if you have saved artifacts with a config and then the c++ type changes/etc.

Reply:
(to clarify - fine for now, and I'm happy to help do the work to convert to an attribute when I get to it - mostly just wanted to stick a placeholder here for us to chat about in the future :)

Reply:
Yes, I was hoping to chat about it too. Let me know when you want to. (I am mostly open rest of the day). I can make this just a `DictionaryAttr`. As I was looking at this, it seemed like a way we could basically "encode" how to lower a dispatch region. Either say "use this known pipeline" or a string that says "use this sequence of passes". I think that might open up some other "easy hackability using IREE" avenues. I am happy to change this to `DictionaryAttr`.